itou/approvals/tests/tests.py

```diff
@@ -1574,15 +1588,23 @@ def test_race_condition(self):
         def first_request():
             nonlocal approval
             with transaction.atomic():
-                approval = ApprovalFactory.build(user=user1, number=Approval.get_next_number())
+                approval = ApprovalFactory.build(
+                    user=user1,
+                    number=Approval.get_next_number(),
+                    eligibility_diagnosis=None,
+                )
                 time.sleep(0.2)  # sleep long enough for the concurrent request to start
                 approval.save()
 
         def concurrent_request():
             nonlocal approval2
             with transaction.atomic():
                 time.sleep(0.1)  # ensure we are not the first to take the lock
-                approval2 = ApprovalFactory.build(user=user2, number=Approval.get_next_number())
+                approval2 = ApprovalFactory.build(
+                    user=user2,
+                    number=Approval.get_next_number(),
+                    eligibility_diagnosis=None,
```

Comment:
pour ma gouverne, pourquoi mettre tous ces None ? On pourrait aussi bien laisse la diagnsotic auto-généré désormais par la Factory ? Qu'est-ce qui bloque ?

Reply:
On a des erreurs du type `<PrescriberOrganization: Denise Devaux>" needs to have a value for field "id" before this many-to-many relationship can be used.`